index.bs

```diff
@@ -2916,6 +2916,214 @@ value and terminate the operation.
 
 </div>
 
+### Signal Credential Changes to the Authenticator - PublicKeyCredential's `signal()` Method {#sctn-signal}
+
+<xmp class="idl">
+    partial interface PublicKeyCredential {
+        static Promise<undefined> signal(PublicKeyCredentialSignalOptions options);
+    };
+
+    dictionary PublicKeyCredentialSignalOptions {
+        USVString                              rpId;
+        Base64URLString                        unknownCredentialId;
+        PublicKeyCurrentCredentialDetails      currentCredentials;
+    };
+
+    dictionary PublicKeyCurrentCredentialDetails {
+        required Base64URLString               userId;
+        PublicKeyCurrentCredentialUserDetails  user;
+        sequence<Base64URLString>              allAcceptedCredentialIds;
+    };
+
+    dictionary PublicKeyCurrentCredentialUserDetails {
+        required DOMString                     name;
+        required DOMString                     displayName;
+    };
+</xmp>
+
+[=[WRPS]=] may use {{PublicKeyCredential/signal()}} to signal [=authenticators=]
+the state of [=public key credentials=], so that incorrect or revoked
+credentials may be updated or removed. [=User agents=] provide this
+functionality opportunistically, since an authenticator may not support updating
+its [=credentials map=] or it may not be attached at the time the request is
+made. Furthermore, in order to avoid revealing information about a user's
+credentials without [=user consent=], the {{PublicKeyCredential/signal()}}
+operation does indicate whether the operation succeeded. A successfully resolved
+promise only indicates that the {{PublicKeyCredentialSignalOptions}} object was
+well formed and accepted.
+
+A number of <dfn>credential reports</dfn> can be signalled by [=[WRPS]=]. Each
+[=credential report=] declares a <dfn>validation algorithm</dfn> and <dfn>
+authenticator actions</dfn>. The input for both these algorithms is the
+{{PublicKeyCredentialSignalOptions}} |options|.
+
+#### <dfn>unknownCredentialId</dfn> report #### {#sctn-unknownCredentialIdReport}
+
+Signals that a [=credential id=] was not recognized by the [=[WRP]=], e.g.
+because it was deleted by the user.
+
+The [=unknownCredentialId=] [=validation algorithm=] is as follows:
+1. If the result of [=base64url encoding | base64url decoding=]
+    <code>|options|.{{PublicKeyCredentialSignalOptions/unknownCredentialId}}</code>
+    is an error, then throw a {{TypeError}}.
+    
+The [=unknownCredentialId=] [=authenticator actions=] are as follows:
+1. [=map/For each=] [=public key credential source=] |credential| in the
+    [=authenticator=]'s [=credential map=]:
+    1. If the |credential|'s [=public key credential source/rpId=] does
+        not equal
+        <code>|options|.{{PublicKeyCredentialSignalOptions/rpId}}</code>,
+        [=continue=].
```

Comment:
Could we drop this `rpId` for the `unknownCredentialId` report? Does the client need it when it finds the authenticator that recognizes the credential ID so that the authenticator can actually delete the right credential? If this is about dealing with potential collisions in credential IDs for different RP IDs I can see a benefit to keeping `rpId`, but the way I see it if we can embrace [the "probabilistically-unique" property of credential IDs](https://pr-preview.s3.amazonaws.com/nsatragno/webauthn/pull/2093.html#credential-id) then could we make it simpler for RPs to signal this kind of report.

Reply:
We need some form of `rpId` validation, otherwise a relying party can remove passkeys from another relying party that is unaffiliated with it if it knows the (~public) credential id.

Reply:
I agree this must be restricted to the caller's own credentials. I suppose RP ID _could_ be optional with a default like for `PublicKeyCredentialRequestOptions.rpId`, but it's good practice to set that `rpId` explicitly anyway and the same would apply here, so making it optional is only mildly useful.